src/server.js

```diff
@@ -1,168 +1,194 @@
-});
-const serverPort = process.env.SCREENIE_PORT || 3000;
-
+logger.verbose('Setting up ENVS');
+const chromiumArgs = process.env.SCREENIE_CHROMIUM_ARGS
+  ? { args: process.env.SCREENIE_CHROMIUM_ARGS.split(' ') }
+  : {};
+const chromiumExec = process.env.SCREENIE_CHROMIUM_EXEC
+  ? { executablePath: process.env.SCREENIE_CHROMIUM_EXEC }
+  : {};
+const defaultFormat = process.env.SCREENIE_DEFAULT_FORMAT || 'jpeg';
 const imageSize = {
-    width: process.env.SCREENIE_WIDTH || 1024,
-    height: process.env.SCREENIE_HEIGHT || 768,
+  width: process.env.SCREENIE_WIDTH || 1024,
+  height: process.env.SCREENIE_HEIGHT || 768,
 };
-const supportedFormats = [
-    'gif',
-    'jpeg',
-    'jpg',
-    'pdf',
-    'png',
-];
-const defaultFormat = process.env.SCREENIE_DEFAULT_FORMAT || 'jpeg';
+const serverPort = process.env.SCREENIE_PORT || 3000;
+const supportedFormats = ['jpg', 'jpeg', 'pdf', 'png'];
+
+const app = new Koa();
+logger.verbose('Created KOA server');
+const pool = createPuppeteerPool({
+  min: process.env.SCREENIE_POOL_MIN || 2,
+  max: process.env.SCREENIE_POOL_MAX || 10,
+  puppeteerArgs: Object.assign({}, chromiumArgs, chromiumExec),
+});
+logger.verbose('Created Puppeteer POOL');
 
 /*
- * Clean up the PhantomJS pool before exiting when receiving a termination
+ * Clean up the Puppeteer pool before exiting when receiving a termination
  * signal. Exit with status code 143 (128 + SIGTERM's signal number, 15).
  */
 process.on('SIGTERM', () => {
-    logger.info('Received SIGTERM, exiting...');
-    pool.drain()
-        .then(() => pool.clear())
-        .then(() => process.exit(143));
+  logger.info('Received SIGTERM, exiting...');
+  pool
+    .drain()
+    .then(() => pool.clear())
+    .then(() => process.exit(143));
 });
 
 /**
- * Set up a PhantomJS instance with a page and configure viewport size.
+ * Set up a Puppeteer instance for a page and configure viewport size.
  */
-app.use(function *(next) {
-    const size = {
-        width: Math.min(
-            2048,
-            parseInt(this.request.query.width, 10) || imageSize.width
-        ),
-        height: Math.min(
-            2048,
-            parseInt(this.request.query.height, 10) || imageSize.height
-        ),
-    };
-
-    logger.verbose(
-        `Instantiating PhantomJS with page size ${size.width}x${size.height}`
-    );
-
-    yield pool.use(instance => instance.createPage())
-        .then(page => this.state.page = page)
-        .then(() => this.state.page.property('viewportSize', size));
-
-    yield next;
+app.use(function*(next) {
+  const { width, height } = this.request.query;
+  const size = {
+    width: Math.min(2048, parseInt(width, 10) || imageSize.width),
+    height: Math.min(2048, parseInt(height, 10) || imageSize.height),
+  };
+  let pageError;
+
+  logger.verbose(`Instantiating Page with size ${size.width}x${size.height}`);
+
+  yield pool.use(instance => {
+    logger.verbose('Use pool and create page.');
```

Comment:
Maybe more useful to say something like `Using browser instance with PID N`. Also, should be consistent in usage of period or not at the end of log messages. I vote for no period.